aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -58,13 +58,29 @@ Tensor& add_(Tensor& self, const Tensor& other, Scalar alpha) {
 }
 
 Tensor& div_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  if (isIntegralType(result.scalar_type(), /*includeBool=*/ true)) {
+    TORCH_WARN_ONCE(
+      "Integer division of tensors using div or / is deprecated. ",
+      "In a future release div and / will ",
```

Comment:
nit: I'd just specify the expected versions, I don't see why we'd deviate from the plan and even if we do, it's not the biggest deal.

Reply:
Also, did we talk about having a future true_division for PyTorch?  (So the error message could say "or import...").

Reply:
I'll put up a separate PR shortly for that behavior.

Reply:
See https://github.com/pytorch/pytorch/pull/34794